### PR TITLE
add limiting comments feature

### PR DESCRIPTION
**Pull request by shmcaffrey**

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -71,4 +82,32 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
 
     response.sendRedirect("/index.html");
   }
+
+    private String convertJsonUsingGson(ArrayList<String> comments) {
```
Comment (robotsaurus) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: nit: looks like this is indented more than the others

File: portfolio/src/main/webapp/script.js [hunk 2]
```diff
@@ -66,11 +66,15 @@ function revealPose () {
...
-
+function getComments(query) {
+    if (query == null) {
+        query == "5";
+    }
     // fetch data from server, parse as json file, then reference coms as an object
-    fetch('/data').then(response => response.json()).then((comments) => {
+    fetch("/data?num-comments=" + query).then(response => response.json()).then((comments) => {
         const commentList = document.getElementById("comment-container");
+        console.log("deleting old comments");
```
Comment (robotsaurus) [inline portfolio/src/main/webapp/script.js]: this doesn't seem to be appropriate here

File: portfolio/src/main/webapp/index.html [hunk 3]
```diff
@@ -72,13 +72,19 @@ <h1 style="font-size:5vw">Shaunna Caffrey</h1>
...
+            <input type="number" id="num-comments" name="num-comments" min="0" value="5" onchange="getComments(this.value)">
+           </form>
+           <form action="/data" method="POST" style="color:black">
+            <h1>Enter your comment:</h1>
+            <textarea name="text-input" placeholder="Enter comment here..."></textarea>
             <br/>
             <input type="submit"/>
           </form>
+          <ul id="comment-container"></ul>
+          <!-- <button onclick="deleteAllComments()">Delete All Comments</button> -->
```
Comment (robotsaurus) [inline portfolio/src/main/webapp/index.html]: Is this meant to be commented out?

**Review by robotsaurus — CHANGES_REQUESTED** [review]

File: portfolio/src/main/webapp/script.js [hunk 4]
```diff
@@ -86,3 +90,7 @@ function createListElement(comment) {
   return liElement;
 }
 
+// Once data from datastore is deleted then remove text from page
+function deleteAllComments() {
+    fetch("/delete-data", {method: 'POST'}).then(document.getElementById("comment-contianer").innerText = "");
```
Comment (snazzorama) [inline portfolio/src/main/webapp/script.js]: nit: "container" is misspelled

**Comment by shmcaffrey** [comment]
ptal

**Review by snazzorama — APPROVED** [review]

**Comment by shmcaffrey** [comment]
ptal

**Review by robotsaurus — APPROVED** [review]

**Pull request merged**